### ISSUE TITLE
Orphaned files when camera renamed

### ISSUE THREAD

**Issue by Whytey** [blakeblackshear/frigate#1279]
**Describe the bug**
After having a camera configured temporarily (for testing purposes) I have noticed that there are old clips and recordings that should have been purged by now that haven't been.

**Version of frigate**
0.8.4-5043040

**Config file**
Can be added, but important part is...
```yaml
clips:
  tmpfs_cache_size: 256m
  retain:
    default: 10
record:
  enabled: True
  retain_days: 7
```

**Frigate stats**
```json
{
  "backyard": {
    "camera_fps": 5.0, 
    "capture_pid": 67, 
    "detection_fps": 0.0, 
    "pid": 59, 
    "process_fps": 5.0, 
    "skipped_fps": 0.0
  }, 
  "courtyard": {
    "camera_fps": 5.1, 
    "capture_pid": 70, 
    "detection_fps": 0.0, 
    "pid": 60, 
    "process_fps": 5.1, 
    "skipped_fps": 0.0
  }, 
  "detection_fps": 0.1, 
  "detectors": {
    "cpu1": {
      "detection_start": 0.0, 
      "inference_speed": 70.71, 
      "pid": 52
    }, 
    "cpu2": {
      "detection_start": 0.0, 
      "inference_speed": 72.47, 
      "pid": 54
    }
  }, 
  "driveway": {
    "camera_fps": 5.0, 
    "capture_pid": 75, 
    "detection_fps": 0.0, 
    "pid": 61, 
    "process_fps": 5.0, 
    "skipped_fps": 0.0
  }, 
  "garage": {
    "camera_fps": 12.2, 
    "capture_pid": 64, 
    "detection_fps": 0.1, 
    "pid": 57, 
    "process_fps": 12.2, 
    "skipped_fps": 0.0
  }, 
  "gate": {
    "camera_fps": 12.0, 
    "capture_pid": 65, 
    "detection_fps": 0.0, 
    "pid": 58, 
    "process_fps": 12.0, 
    "skipped_fps": 0.0
  }, 
  "patio": {
    "camera_fps": 5.1, 
    "capture_pid": 80, 
    "detection_fps": 0.0, 
    "pid": 62, 
    "process_fps": 5.1, 
    "skipped_fps": 0.0
  }, 
  "pool": {
    "camera_fps": 5.1, 
    "capture_pid": 83, 
    "detection_fps": 0.0, 
    "pid": 63, 
    "process_fps": 5.1, 
    "skipped_fps": 0.0
  }, 
  "service": {
    "storage": {
      "/dev/shm": {
        "free": 58.9, 
        "mount_type": "tmpfs", 
        "total": 67.1, 
        "used": 8.2
      }, 
      "/media/frigate/clips": {
        "free": 2236633.1, 
        "mount_type": "ext4", 
        "total": 3937851.5, 
        "used": 1501162.3
      }, 
      "/media/frigate/recordings": {
        "free": 2236633.1, 
        "mount_type": "ext4", 
        "total": 3937851.5, 
        "used": 1501162.3
      }, 
      "/tmp/cache": {
        "free": 740.2, 
        "mount_type": "tmpfs", 
        "total": 1000.0, 
        "used": 259.8
      }
    }, 
    "uptime": 433288, 
    "version": "0.8.4-5043040"
  }
}
```

**Computer Hardware**
 - OS: Ubuntu LXC container
 - Install method: Docker under LXC :)
 - Virtualization: LXC
 - Coral Version: None - CPUs
 - Network Setup: Wired

**Camera Info:**
N/A

**Additional context**
A couple of clips that are hanging around...
```
root@frigate:/data/frigate/clips# ls -al *new*
-rw-r--r-- 1 root root   20934 May  6 23:01 new_gate-1620306046.75046-6kr010.jpg
-rw-r--r-- 1 root root 8162921 May  6 23:01 new_gate-1620306046.75046-6kr010.mp4
-rw-r--r-- 1 root root   24894 May 26 09:35 new_gate-1621985675.373239-j3ic3j.jpg
```

A small sample of recordings that are hanging around, also...
```
root@frigate:/data/frigate/recordings/2021-05# ls -R
.:
29  30  31

./29:
20  21  22  23

./29/20:
new_gate

./29/20/new_gate:
27.12.mp4  29.12.mp4  31.12.mp4  33.12.mp4  35.12.mp4  37.12.mp4  39.12.mp4  41.12.mp4  43.12.mp4  45.12.mp4  47.12.mp4  49.12.mp4  51.12.mp4  53.12.mp4  55.12.mp4  57.12.mp4  59.12.mp4
28.12.mp4  30.12.mp4  32.12.mp4  34.12.mp4  36.12.mp4  38.12.mp4  40.12.mp4  42.12.mp4  44.12.mp4  46.12.mp4  48.12.mp4  50.12.mp4  52.12.mp4  54.12.mp4  56.12.mp4  58.12.mp4

./29/21:
new_gate

./29/21/new_gate:
00.12.mp4  04.12.mp4  08.12.mp4  12.12.mp4  16.12.mp4  20.12.mp4  24.12.mp4  28.12.mp4  32.12.mp4  36.12.mp4  40.12.mp4  44.12.mp4  48.12.mp4  52.12.mp4  56.12.mp4
01.12.mp4  05.12.mp4  09.12.mp4  13.12.mp4  17.12.mp4  21.12.mp4  25.12.mp4  29.12.mp4  33.12.mp4  37.12.mp4  41.12.mp4  45.12.mp4  49.12.mp4  53.12.mp4  57.12.mp4
02.12.mp4  06.12.mp4  10.12.mp4  14.12.mp4  18.12.mp4  22.12.mp4  26.12.mp4  30.12.mp4  34.12.mp4  38.12.mp4  42.12.mp4  46.12.mp4  50.12.mp4  54.12.mp4  58.12.mp4
03.12.mp4  07.12.mp4  11.12.mp4  15.12.mp4  19.12.mp4  23.12.mp4  27.12.mp4  31.12.mp4  35.12.mp4  39.12.mp4  43.12.mp4  47.12.mp4  51.12.mp4  55.12.mp4  59.12.mp4```

**Comment by Whytey** [comment]
Suggestion would be to perhaps have a watchdog/auditor that cleansup the old files in the ```/media/frigate``` folder.  I think you should be safe to assume that frigate has full control over that folder and what should be saved/deleted.

**Comment by gpete** [comment]
This same thing happened to me and I was surprised to find that there is a feature to cleanup these files but it wasn't working. Turns out there was a bug and I push a fix for it (#1026). It looks like it should go out in the 0.9.0 release, but I'm not sure when that will go out. 

Interestingly enough the bug will still let some files get cleaned up on each pass (which happens every 5 minutes). You might find that they will eventually all get cleaned up if left long enough.
